### PR TITLE
Allow overriding ensure_ascii in the "to_json" template filter 

### PR DESCRIPTION
**Pull request by koying**
## Proposed change

Allow overriding ensure_ascii in the to_json template filter

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/54527
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -357,6 +357,8 @@ To fix it, enforce the ISO conversion via `isoformat()`:
 
 The `to_json` filter serializes an object to a JSON string. In some cases, it may be necessary to format a JSON string for use with a webhook, as a parameter for command-line utilities or any number of other applications. This can be complicated in a template, especially when dealing with escaping special characters. Using the `to_json` filter, this is handled automatically.
 
+Similarly to the python equivalent, the filter accepts a `ensure_ascii` parameter, defaulting to `True`, which specifies the encoding of the resulting JSON: UTF-8 or escaped ASCII.
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Similarly to the python equivalent, the filter accepts an `ensure_ascii` parameter, defaulting to `True`. If `ensure_ascii` is `True`, the output is guaranteed to have all incoming non-ASCII characters escaped. If `ensure_ascii` is false, these characters will be output as-is.
```

Comment (koying) [reply source/_docs/configuration/templating.markdown]: "as-is" does not mean much in technical terms ;)
But if you think it's clearer to users, fine by me.

Comment (emontnemery) [reply source/_docs/configuration/templating.markdown]: The suggestion is copied verbatim from the official Python documentation.
If you don't like "as-is", "unchanged" would be fine too.

Comment (koying) [reply source/_docs/configuration/templating.markdown]: Nah, as-is is clear enough for non-techies

**Review by emontnemery — APPROVED** [review]
Thanks, @koying 👍

**Pull request merged**